### DISCUSSION TITLE
Need help for Drift autoupdate listview with watch()

### DISCUSSION THREAD

**Discussion by ghost** [simolus3/drift#3012] [Q&A] [answered]
I'm new to flutter, I'm building an app that should show a list of itmes (Articoli), and from other pages I can add edit or remove them. It worked, but when I made change I need to restart the app to view the changes in the list, so I looked in the documentation and I found watch(), I tryed for the last 3 day to make it work, but I was not able, now it show the circularindicator forever, and no items in the list. here is my code https://pastebin.com/cwisd4ed  anyone can help me to find my mistakes?

**Comment by simolus3** [comment, marked as answer]
I think `ConnectionState.done` is terminal and signals that the stream has completed, with no subsequent updates to be expected. As query streams don't end, you'll never reach that state.
Instead, you should check for `ConnectionState.active` or just show the state depending on whether `hasData` or `hasError` is true on the snapshot.

**Reply by ghost** [reply]
Thanks it worked! 
I've another question:
I've declare this tables:

```
import 'package:drift/drift.dart';
 
class items extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get description => text().named('description')();
 
}
class Orders extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get description => text().named('description')();
  TextColumn get items => text().named('items')();
  IntColumn get quantity => integer().named('quantity')();
}

```
but in Orders table items is text, instead of "items" since I have not "itemsColumn" type, how should I do?

**Reply by simolus3** [reply]
I'm not sure I understand - `named` changes the name of the column in SQL. By default, drift will use the `snake_case` for the name in SQL. So columns that aren't more than one word, it doesn't make a difference. `TextColumn get description` is `description TEXT` in SQL with and without the `.named()`. Only if you have something like `quantityOfItems` does it make a difference, since that is `quantity_of_items` in SQL by default.

> but in Orders table items is text, instead of "items" since I have not "itemsColumn" type

I suppose you want to reference multiple items in an order? The "clean" relational way to do that is to use three tables:

1. Items
2. Orders
3. A table combining the two, e.g. `ItemsInOrders`. This would store one row for each item part of an order.

You can also store a list of item numbers directly in the `Orders` table without using a third table. You'd still use a text column for that though. In addition, you'd add a type converter converting the column into a `List<int>` in Dart as [shown in this example](https://drift.simonbinder.eu/docs/examples/relationships/#with-json-functions).
